File: 2392-successful-pairs-of-spells-and-potions/successful-pairs-of-spells-and-potions.py
class Solution:
    def successfulPairs(self, spells: List[int], potions: List[int], success: int) -> List[int]:
        potions.sort()
        m = len(potions)
        ans = []
        for i in spells:
            ans.append(m-bisect_left(potions,success/i))
        return ans

File: 2392-successful-pairs-of-spells-and-potions/successful-pairs-of-spells-and-potions.cpp
class Solution {
public:
    vector<int> successfulPairs(vector<int>& spells, vector<int>& potions, long long success) {
        sort(potions.begin(), potions.end());
        vector<int> ans;
        int m = potions.size();
        for (int i: spells){
            int c = (potions.end()-lower_bound(potions.begin(), potions.end(), (success+i-1)/i));
            ans.push_back(c);
        }
        return ans;
    }
};